app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ShoppingListController < ApplicationController
+  def index
+    @recipes = Recipe.where(user_id: current_user.id)
+    @ingredients = RecipeFood.where(recipe_id: @recipes.map(&:id))
+    @foods = Food.where(id: @ingredients.map(&:food_id))
+  end
+
+  def total_price
+    price = 0
+    @ingredients.each do |recipe_food|
+      price += Food.find(recipe_food.food_id).price * recipe_food.quantity
+    end
+    price
+  end
+
+  helper_method :total_price
+end
```

Comment:
- [ ] Kindly move this method that calculates the price to the `recipe food` model so that we keep only the necessary logic inside the controller and put everything else in models (thin controllers rule).